docs/source/configuration.rst

```diff
@@ -273,15 +285,51 @@ will allow 10 messages at once before triggering the throttle mechanism, then
 it'll wait 0.5s before sending a new message, and refill the burst limit every
 2 seconds.
 
+The wait time **cannot be longer** than :attr:`~CoreSection.flood_max_wait` (2s
+by default). This maximum wait time includes any potential extra penalty for
+longer messages.
+
+Messages that are longer than :attr:`~CoreSection.flood_text_length` get an
+extra wait penalty. The penalty is computed using a penalty ratio (controlled
+by :attr:`~CoreSection.flood_penalty_ratio`, which is 1.4 by default)::
+
+    length_overflow = max(0, (len(text) - flood_text_length))
+    extra_penalty = length_overflow / (flood_text_length * flood_penalty_ratio)
+
+For example with a message of 80 characters, the added extra penalty will be::
+
+    length_overflow = max(0, 80 - 50)  # == 30
+    extra_penalty = 30 / (50 * 1.4)  # == 0.428s (approximately)
+
+With the default configuration, it means a minimum wait time of 0.928s before
+sending any new message (0.5s + 0.428s).
+
+You can **deactivate** this extra wait penalty by setting
+:attr:`~CoreSection.flood_penalty_ratio` to 0 (or less).
+
 The default configuration works fine with most tested networks, but individual
 bots' owners are invited to tweak as necessary to respect their network's flood
 policy.
 
 .. versionadded:: 7.0
 
-   Flood prevention has been modified in Sopel 7.0 and these configuration
-   options have been added: ``flood_burst_lines``, ``flood_empty_wait``, and
-   ``flood_refill_rate``.
+    Flood prevention has been modified in Sopel 7.0 and these configuration
+    options have been added: ``flood_burst_lines``, ``flood_empty_wait``, and
+    ``flood_refill_rate``.
+
+.. versionadded:: 7.1
+
+    Even more options have been added in Sopel 7.1: ``flood_max_wait``,
+    ``flood_text_length``, and ``flood_penalty_ratio``.
+
+    It is now possible to deactivate the extra penalty for longer messages by
+    setting ``flood_penalty_ratio`` to 0.
```

Comment:
Mentioning the version again in the body text of a `versionadded` block is redundant.

```suggestion
    Additional configuration options: ``flood_burst_lines``, ``flood_empty_wait``,
    and ``flood_refill_rate``.

.. versionadded:: 7.1

    Even more additional configuration options: ``flood_max_wait``,
    ``flood_text_length``, and ``flood_penalty_ratio``.

    It is now possible to deactivate the extra penalty for longer messages by
    setting ``flood_penalty_ratio`` to 0.
```